### PR TITLE
Bump the MSRV to 1.63 

### PR DESCRIPTION
**Pull request by Michael-F-Bryan**
We upgraded our dependencies as part of #3254 (https://github.com/wasmerio/wasmer/commit/9b13fae93e6da80eebf335fa2d86fbd3e210a6c7), which pulled in version `1.2.0` of the `derive_arbitrary` crate. This requires Rust `1.63` to compile, but our current minimum supported Rust version is `1.61`. 

This PR bumps our MSRV to match that of `derive_arbitrary`.

# Review

- [ ] Add a short description of the change to the CHANGELOG.md file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -159,9 +161,9 @@ Looking for changes that affect our C API? See the [C API Changelog](lib/c-api/C
 - [#2943](https://github.com/wasmerio/wasmer/pull/2943) Fix build error on some archs by using c_char instead of i8
 - [#2976](https://github.com/wasmerio/wasmer/pull/2976) Upgrade minimum enumset to one that compiles
 - [#2988](https://github.com/wasmerio/wasmer/pull/2988) Have make targets install-capi-lib,install-pkgconfig work without building the wasmer binary
-- [#2967](https://github.com/wasmerio/wasmer/pull/2967) Fix singlepass on arm64 that was trying to emit a sub opcode with a constant as destination (for #2959) 
+- [#2967](https://github.com/wasmerio/wasmer/pull/2967) Fix singlepass on arm64 that was trying to emit a sub opcode with a constant as destination (for #2959)
 - [#2948](https://github.com/wasmerio/wasmer/pull/2948) Fix regression on gen_import_call_trampoline_arm64()
-- [#2944](https://github.com/wasmerio/wasmer/pull/2944) Fix duplicate entries in the CHANGELOG 
+- [#2944](https://github.com/wasmerio/wasmer/pull/2944) Fix duplicate entries in the CHANGELOG
```
Comment (ptitSeb) [inline CHANGELOG.md:167]: The hunt of the trailling space :P !

File: lib/vm/src/store.rs [hunk 2]
```diff
@@ -273,12 +273,12 @@ where
...
+            Self::Host(p) => {
                 write!(f, "host(")?;
                 p.as_ref().fmt(f)?;
                 write!(f, ")")
             }
-            MaybeInstanceOwned::Instance(p) => {
+            Self::Instance(p) => {
                 write!(f, "instance(")?;
                 unsafe { p.as_ref().fmt(f)? };
                 write!(f, ")")
```
Comment (ptitSeb) [inline lib/vm/src/store.rs:284]: I always forgot to fix thoses one. It's done now, good.

File: tests/compilers/typed_functions.rs [hunk 3]
```diff
@@ -244,11 +244,11 @@ fn static_host_function_without_env(config: crate::Config) -> anyhow::Result<()>
     let mut store = config.store();
 
     fn f(a: i32, b: i64, c: f32, d: f64) -> (f64, f32, i64, i32) {
-        (d * 4.0, c * 3.0, b * 2, a * 1)
+        (d * 4.0, c * 3.0, b * 2, a)
     }
```
Comment (ptitSeb) [inline tests/compilers/typed_functions.rs:248]: Not a fan of the "* 1" removal, as it was here for clarity. Still works, but, meh

**Review by ptitSeb — APPROVED** [review]

File: tests/compilers/typed_functions.rs (hunk 3 above)
Comment (Michael-F-Bryan) [reply tests/compilers/typed_functions.rs:248]: Yeah, it looks like clippy warns you about redundant multiplies so `clippy --fix` will "help" out by removing them :disappointed:

**Pull request merged**